data-prepper-plugins/flatten-json-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/flattenjson/FlattenJsonProcessorTest.java

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor.flattenjson;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.opensearch.dataprepper.expression.ExpressionEvaluator;
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.event.JacksonEvent;
+import org.opensearch.dataprepper.model.record.Record;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class FlattenJsonProcessorTest {
+    private static final String SOURCE_KEY = "source";
+    private static final String TARGET_KEY = "target";
+
+    @Mock
+    private PluginMetrics pluginMetrics;
+
+    @Mock
+    private FlattenJsonProcessorConfig mockConfig;
+
+    @Mock
+    private ExpressionEvaluator expressionEvaluator;
+
+    @BeforeEach
+    void setUp() {
+        lenient().when(mockConfig.getSource()).thenReturn("");
+        lenient().when(mockConfig.getTarget()).thenReturn("");
+        lenient().when(mockConfig.isRemoveProcessedFields()).thenReturn(false);
+        lenient().when(mockConfig.isRemoveListIndices()).thenReturn(false);
+        lenient().when(mockConfig.getFlattenWhen()).thenReturn(null);
+        lenient().when(mockConfig.getTagsOnFailure()).thenReturn(new ArrayList<>());
+    }
+
+    @Test
+    void testFlattenEntireEventData() {
+        final FlattenJsonProcessor processor = createObjectUnderTest();
+        final Record<Event> testRecord = createTestRecord(createTestData());
+        final List<Record<Event>> resultRecord = (List<Record<Event>>) processor.doExecute(Collections.singletonList(testRecord));
+
+        assertThat(resultRecord.size(), is(1));
+
+        final Event resultEvent = resultRecord.get(0).getData();
+        Map<String, Object> resultData = resultEvent.get("", Map.class);
+
+        assertFlattenedData(resultData);
+
+        assertThat(resultData.containsKey("key2"), is(true));
```

Comment:
Is key1 deleted here? Can you add assertion?

Reply:
key1 is checked as part of `assertFlattenedData(resultData)` (one line above)